GLE_solver_v4.py

```
@@ -0,0 +1,331 @@
+    # Solve the ODEs
+    try:
+        solution = solve_bvp(GLE, boundary_conditions, s_range_local, y_guess_local, max_nodes=1000000)
+        
+        if solution.success:
+            # Return h(s_max) - h_end (should be zero when correct s_max is found)
+            h_final = solution.y[0, -1]
+            return h_final - h_end
+        else:
+            # If solution failed, return a large penalty
+            return 1e6
+    except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)